### ISSUE TITLE
uninitialized constant Chef::Resource::HomebrewTap Chef 13.0.118

### ISSUE THREAD

**Issue by dombialcz** [chef-cookbooks/homebrew#120]
### Cookbook version
3.0.0

### Chef-client version
13.0.118

### Platform Details
mac_os_x 10.12.2 (Sierra)

### Scenario:
install java using java cookbook, but homebrew recipe fails even before that.

### Steps to Reproduce:
include_recipe 'java::default'

source: [java cookbook: homebrew recipe](https://github.com/agileorbit-cookbooks/java/blob/master/recipes/homebrew.rb)
```
include_recipe 'homebrew'
include_recipe 'homebrew::cask'
include_recipe 'java::notify'

homebrew_tap 'caskroom/versions'
homebrew_cask "java#{node['java']['jdk_version']}" do
  notifies :write, 'log[jdk-version-changed]', :immediately
end
```

### Expected Result:
java installed

### Actual Result:

**Error executing action `tap` on resource 'homebrew_tap[caskroom/cask]'**

------------

> ```
> Recipe: homebrew::cask
>          * homebrew_tap[caskroom/cask] action tap[2017-04-18T08:36:24+00:00] INFO: Processing homebrew_tap[caskroom/cask] action tap (homebrew::cask line 20)
>        [2017-04-18T08:36:24+00:00] INFO: Processing homebrew_tap[caskroom/cask] action tap (homebrew::cask line 20)
>        
>            
>            ================================================================================
>            Error executing action `tap` on resource 'homebrew_tap[caskroom/cask]'
>            ================================================================================
>            
>            NameError
>            ---------
>            uninitialized constant Chef::Resource::HomebrewTap
>            
>            Cookbook Trace:
>            ---------------
>            /tmp/kitchen/cache/cookbooks/homebrew/providers/tap.rb:27:in `load_current_resource'
>            
>            Resource Declaration:
>            ---------------------
>            # In /tmp/kitchen/cache/cookbooks/homebrew/recipes/cask.rb
>            
>             20: homebrew_tap 'caskroom/cask'
>             21: 
>            
>            Compiled Resource:
>            ------------------
>            # Declared in /tmp/kitchen/cache/cookbooks/homebrew/recipes/cask.rb:20:in `from_file'
>            
>            homebrew_tap("caskroom/cask") do
>              action [:tap]
>              default_guard_interpreter :default
>              declared_type :homebrew_tap
>              cookbook_name "homebrew"
>              recipe_name "cask"
>            end
>            
>            System Info:
>            ------------
>            chef_version=13.0.118
>            platform=mac_os_x
>            platform_version=10.12.2
>            ruby=ruby 2.4.1p111 (2017-03-22 revision 58053) [x86_64-darwin14]
>            program_name=chef-client worker: ppid=920;start=08:31:53;
>            executable=/opt/chef/bin/chef-client
>            
> 
> 
> ```

**Comment by dombialcz** [comment]
Submitted a fix here: [](https://github.com/chef-cookbooks/homebrew/pull/121)

**Comment by rjaros87** [comment]
:+1:

**Comment by tas50** [comment]
Fixed in #119
